### PR TITLE
User search component

### LINKED ISSUE #32: User Search
**Describe the task that needs to be done.**
A component for searching for the user/group to talk to.

**Describe how a solution to your proposed task might look like (and any alternatives considered).**
Would expect an edit text box that refreshes a search each time there's a change to the text (so some sort of custom method attached to onChange)

**Notes**

### PR DESCRIPTION
**Pull request by sv-singh**
## Description
This component allows users to type in the names of other users for searching.

Fixes #32 

## How has this been tested?
Tests are written to make sure each element is getting rendered correctly i.e. the search icon and placeholder text. There are also tests for ensuring the search icon is working correctly on click and also that the search updates correctly when entering text.

## Screenshots
image

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/components/UserSearch/__tests__/UserSearch.test.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    test('check that handle search works after clicking search icon', () => {
+        jest.spyOn(console, 'log');
+        expect(console.log.mock.calls.length).toBe(0);
+        render(
+            <UserSearch />
+        );
+        userEvent.click(screen.getByTestId('searchIcon'));
+        expect(console.log.mock.calls.length).toBe(1);
+        expect(console.log.mock.calls[0][0]).toBe("search pressed");
```
Comment (scorpionknifes) [inline src/components/UserSearch/__tests__/UserSearch.test.jsx:47]: set a `jest.func()` and send it in as prop for `submitSearch`

Check if jest.func() is called when search is pressed.

File: src/components/UserSearch/UserSearch.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      sx={{
+      }}
+      noValidate
+      autoComplete="off"
+    >
+      <SearchBox
+        id="outlined-basic"
+        variant="outlined"
+        placeholder="Type a name"
+        onChange={(e) => searchItems(e.target.value)}
```
Comment (scorpionknifes) [inline src/components/UserSearch/UserSearch.jsx:61]:
```suggestion
        onChange={handleSearch}
```
With the above simplifications

**Review by scorpionknifes — CHANGES_REQUESTED** [review]
Great work.

_Couple suggested changes:_
tldr: remove `console.log` for testing, add props and simplications.
Also, remember to set up **prettier** and format your files thx.

File: src/components/UserSearch/UserSearch.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  // log the search input as it's being typed
+  const searchItems = (searchValue) => {
+    setSearchInput(searchValue)
+    console.log(searchInput);
+  }
+
+  // placeholder for the handle search method
+  const handleSearch = () => {
+    console.log('search pressed')
+  };
```
[the remark this answers is not in the document]
Comment (sv-singh) [reply src/components/UserSearch/UserSearch.jsx]: Thanks for the recommendations I've made those changes.

File: src/components/UserSearch/UserSearch.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+// searchbox styling
+const SearchBox = styled(TextField)(() => ({
+  '& fieldset': {
+    borderRadius: '25px',
+    backgroundColor: '#FAFAFC',
+    zIndex: -1,
+  },
+
+  '& .MuiInputBase-input': {
+    width: '250px',
```
Comment (scorpionknifes) [inline src/components/UserSearch/UserSearch.jsx:19]: TODO: need to change this to be responsive. we can change this later when we make the buddy list.

**Review by scorpionknifes — APPROVED** [review]
LGTM

**Review by scorpionknifes — APPROVED** [review]
